orttraining/orttraining/test/python/orttraining_test_model_transform.py

```diff
@@ -0,0 +1,93 @@
+from onnx import numpy_helper
+
+def add_name(model):
+    i = 0
+    for node in model.graph.node:
+       node.name = '%s_%d' %(node.op_type, i)
+       i += 1
+
+def find_output_node(model, arg):
+    result = []
+    for node in model.graph.node:
+        for input in node.input:
+            if input == arg:
+                result.append(node)
+    return result[0] if len(result) == 1 else None
```

Comment:
This is finding the one single node that is using `arg` as input.
If `arg` is not used as input by any node, or if there are more than 1 node using `arg` as input, this method returns `None`. Is this expected? What is the use case for this method?

Reply:
details commented in code. In short, a transpose node can be folded only if it receives a weight that does not feed other nodes.